### PR TITLE
Print error and don't exit

### PR DESCRIPTION
**Pull request by benschinn**
We are seeing this error occur:

```
Error getting pg_current_wal_lsn: sql: no rows in resulting set
```

Then pgreba exits with status 1.

Removing `Fatalln` should fix this issue.

cc: @skunkworker @film42

### CODE REVIEW

File: data_source.go [hunk 1]
```diff
@@ -259,12 +259,12 @@ FROM
 
 	pgCurrentWalLsn, err := ds.getPgCurrentWalLsn(ds.cfg.MaxHop, db)
 	if err != nil {
-		log.Fatalln("Error getting pg_current_wal_lsn:", err)
+		log.Println("Error getting pg_current_wal_lsn:", err)
```
Comment (ReUhssurance) [inline data_source.go:262]: If we experience an error in these sections, should we return from the function still?  I'm wondering about potential panics further down (ie. a reference to `pgCurrentWalLsn` where it happens to be nil).

Comment (film42) [reply data_source.go:262]: Yup. We should probably print but add `return nil, err`.

Comment (benschinn) [reply data_source.go:262]: https://github.com/film42/pgreba/blob/2d1cd67c202be1dfa97eaf906899157d4a26cd14/main.go#L37-L41

great point, yes I think we should be returning when an err occurs. If the err is returned, a status 500 is returned

Comment (ReUhssurance) [reply data_source.go:262]: That doesn't look like it's returning a 500 more so as a process panic.

I think we should be doing something like...
```
if err != nil {
    http.Error(w, "Some message here", http.StatusInternalServerError)
    return
}
```

Comment (benschinn) [reply data_source.go:262]: fixed!

**Comment by benschinn** [comment]
got rid of the superfluous writing to header. tested in sand.

Before:
```
Aug 03 21:21:46 sb-sand-postgres101 pgreba[26816]: 2021/08/03 21:21:46 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:65)
Aug 03 21:21:46 sb-sand-postgres101 pgreba[26816]: 10.47.33.194 - - [03/Aug/2021:21:21:46 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:21:48 sb-sand-postgres101 pgreba[26816]: 2021/08/03 21:21:48 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:65)
Aug 03 21:21:48 sb-sand-postgres101 pgreba[26816]: 10.47.33.194 - - [03/Aug/2021:21:21:48 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:21:48 sb-sand-postgres101 pgreba[26816]: 2021/08/03 21:21:48 http: superfluous response.WriteHeader call from github.com/gorilla/handlers.(*responseLogger).WriteHeader (handlers.go:65)
Aug 03 21:21:48 sb-sand-postgres101 pgreba[26816]: 10.47.33.195 - - [03/Aug/2021:21:21:48 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:21:48 sb-sand-postgres101 pgreba[26816]: 10.47.33.195 - - [03/Aug/2021:21:21:48 +0000] "GET /primary HTTP/1.0" 200 419
```

Fixed:
```
Aug 03 21:23:52 sb-sand-postgres101 pgreba[25915]: 2021/08/03 21:23:52 Listening on :8000
Aug 03 21:23:52 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:52 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:52 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:52 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:52 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:52 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:52 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:52 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:54 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:54 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:54 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:54 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:54 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:54 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:54 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:54 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:56 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:56 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:56 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:56 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:56 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:56 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:56 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:56 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:58 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:58 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:58 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:23:58 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:23:58 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:58 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:23:58 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:23:58 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:24:00 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:24:00 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:24:00 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:24:00 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:24:00 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:24:00 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:24:00 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:24:00 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:24:02 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:24:02 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
Aug 03 21:24:02 sb-sand-postgres101 pgreba[25915]: 10.47.33.195 - - [03/Aug/2021:21:24:02 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:24:02 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:24:02 +0000] "GET /primary HTTP/1.0" 200 419
Aug 03 21:24:02 sb-sand-postgres101 pgreba[25915]: 10.47.33.194 - - [03/Aug/2021:21:24:02 +0000] "GET /replica?max_allowable_byte_lag=200000000 HTTP/1.0" 503 419
```

**Comment by ReUhssurance** [comment]
LGTM

**Pull request merged**